src/traces/image/plot.js

```diff
@@ -26,8 +26,10 @@ module.exports = function plot(gd, plotinfo, cdimage, imageLayer) {
         var plotGroup = d3.select(this);
         var cd0 = cd[0];
         var trace = cd0.trace;
-        var fastImage = supportsPixelatedImage && !trace._hasZ && trace._hasSource && xa.type === 'linear' && ya.type === 'linear';
-        trace._fastImage = fastImage;
+        var realImage = trace.zsmooth === 'fast' || (
```

Comment:
I don't think we should use a real image if zsmooth is false and the axes are not linear and/or no `source` attribute is defined (otherwise we'd get a blank image!). 

@archmoj could you please review and possibly cherry-pick my commit  https://github.com/plotly/plotly.js/commit/f83e2876fbc82b1e0fe8b7c1fe01165592e83d1b ?

Reply:
Good call. Picked f83e287.